t/cs.t

```diff
@@ -2526,3 +2526,108 @@ SSL server name: <test.com>
 GET /c
 --- error_log
 key: y7KXwBSpVrxtkR0O+bQt+Q==
+
+
+
+=== TEST 38: server handshake response string
+--- http_config eval: $::HttpConfig
+--- config
+    location = /c {
+        content_by_lua_block {
+            local client = require "resty.websocket.client"
+            local wb, err = client:new()
+            local uri = "ws://127.0.0.1:" .. ngx.var.server_port .. "/s"
+            local ok, err, res = wb:connect(uri)
+            if not ok then
+                ngx.say("failed to connect: ", err)
+                return
+            end
+
+            if not res then
+                ngx.say("no response string")
+                return
+            end
+            ngx.say(res)
+        }
+    }
+
+    location = /s {
+        content_by_lua_block {
+            local server = require "resty.websocket.server"
+            local wb, err = server:new()
+            if not wb then
+                ngx.log(ngx.ERR, "failed to new websocket: ", err)
+                return ngx.exit(444)
+            end
+        }
+    }
+--- request
+GET /c
+--- response_body_like
+^HTTP\/1\.1 101 Switching Protocols.*
+--- no_error_log
+[error]
+[warn]
+
+
+
+=== TEST 39: server handshake response string (reused connection)
+--- http_config eval: $::HttpConfig
+--- config
+    location = /c {
+        content_by_lua_block {
+            local client = require "resty.websocket.client"
+            local wb, err = client:new()
+            local uri = "ws://127.0.0.1:" .. ngx.var.server_port .. "/s"
+            local ok, err, res = wb:connect(uri)
+            if not ok then
+                ngx.say("failed to connect: ", err)
+                return
+            end
+
+            if not res then
+                ngx.say("no response string")
+                return
+            end
+
+            if not res:match("^HTTP/1%.1 101 Switching Protocols\r\n") then
+                ngx.say("invalid response string")
+                return
+            end
+
+            ok, err = wb:set_keepalive()
+            if not ok then
+                ngx.say("failed to enable keepalive: ", err)
+                return
+            end
+
+            wb = client:new()
+            ok, err, res = wb:connect(uri)
+            if not ok then
+                ngx.say("second connect failed: ", err)
+                return
+            end
+
+            ngx.say(res)
+        }
+    }
+
+    location = /s {
+        content_by_lua_block {
+            local server = require "resty.websocket.server"
+            local wb, err = server:new()
+            if not wb then
+                ngx.log(ngx.ERR, "failed to new websocket: ", err)
+                return ngx.exit(444)
+            end
+
+            ngx.sleep(1)
```

Comment:
Do we really need this? It feels hacky. In general, `ngx.sleep` must be avoided in the tests (or if absolutely necessary for some hack, then ideally with decimal-point waiting times e.g. `0.1` or `0.2` max).

Reply:
This exists in the keepalive test case to ensure that the `/s` content handler doesn't exit before the `/c` content handler has a chance to re-instantiate the WS client a second time. So in practice I don't think anything ends up being blocked waiting for this `sleep()` to return.

Adding a sleep here was just the least amount of code to get the test case to work, but I can also rework things a little bit so that there's a more meaningful interaction of `/c` and `/s` if that's better.

Reply:
Yes I do understand why. What I am saying is it isn't ideal. Anyway, good enough.